### PR TITLE
Make it possible to disable operations and assertions.

### PR DESCRIPTION
**Pull request by BenBirt**
fixes https://github.com/dataform-co/dataform/issues/326

### CODE REVIEW

File: api/commands/build.ts [hunk 1]
```diff
@@ -115,15 +115,13 @@ export class Builder {
...
-    const tasks = table.disabled
-      ? ([] as dataform.IExecutionTask[])
-      : this.adapter.publishTasks(table, runConfig, tableMetadata).build();
-
     return {
       ...this.toPartialExecutionAction(table),
       type: "table",
       tableType: table.type,
-      tasks,
+      tasks: table.disabled
```
Comment (lewish) [inline api/commands/build.ts:122]: This won't skip things like pushing metadata back to bigquery, but that's not the end of the world

File: core/operation.ts [hunk 2]
```diff
@@ -111,6 +119,11 @@ export class Operation {
       : dataform.ActionHermeticity.NON_HERMETIC;
   }
 
+  public disabled() {
```
Comment (lewish) [inline core/operation.ts:122]: I wonder if we really want to encourage this syntax if the same can be achieved in the config block? I think at some point we should move to these APIs being fully object based, like:

```
publish({
  name: "x",
  query: ctx => ...
}
```

In other words I'd be tempted to leave this out for now

File: core/session.ts [hunk 3]
```diff
@@ -103,20 +103,6 @@ export class Session {
         "Actions may only contain more than one SQL statement if they are of type 'operations'."
       );
     }
-    if (
```
Comment (lewish) [inline core/session.ts:106]: Why remove all these?

Comment (BenBirt) [reply core/session.ts:106]: your excess-property checks catch these - so it's just a double-error.

File: core/operation.ts (hunk 2 above)
Comment (BenBirt) [reply core/operation.ts:122]: I kinda agree, but OTOH i do think API consistency is king and it costs us very little to have these, so I'd prefer to keep them

File: api/commands/build.ts (hunk 1 above)
Comment (BenBirt) [reply api/commands/build.ts:122]: good catch, but actually the run code does correctly skip that code: https://source.cloud.google.com/tada-analytics/github_dataform-co_dataform/+/master:api/commands/run.ts;l=325?q=executeAction

**Pull request merged**